### PR TITLE
Allow extension pages to be reloaded

### PR DESCRIPTION
**Pull request by tim-hellhake**

### CODE REVIEW

File: static/js/extension.js [hunk 1]
```diff
@@ -32,6 +32,18 @@ class Extension {
...
   }
 
+  /**
+   * Load your html page here
+   *
+   * @returns {Promise} Promise which should resolve as soon as the extension
+   *                    is ready to be shown
+   */
+  load() {
+    return new Promise((resolve) => {
```
Comment (mrstegeman) [inline static/js/extension.js:42]: `return Promise.resolve()` would be simpler.

File: static/js/app.js [hunk 2]
```diff
@@ -288,14 +299,28 @@ const App = {
...
+      // wait until the extensions is loaded before showing it
+      extension.load()
+        .then(() => {
+          extension.show(context);
+          this.extensions[extensionId].show(context);
+          this.selectView(
+            `extension-${Utils.escapeHtmlForIdClass(extensionId)}-view`
+          );
+          this.displayedExtension = extensionId;
+        });
```
Comment (mrstegeman) [inline static/js/app.js:312]: You should add a `catch()` that will just redirect to `/things`.

File: static/js/app.js [hunk 3]
```diff
@@ -288,14 +299,28 @@ const App = {
...
-      this.extensions[extensionId].show(context);
-      this.selectView(
-        `extension-${Utils.escapeHtmlForIdClass(extensionId)}-view`
-      );
-      this.displayedExtension = extensionId;
+      const extension = this.extensions[extensionId];
+      // wait until the extensions is loaded before showing it
+      extension.load()
+        .then(() => {
+          extension.show(context);
```
Comment (mrstegeman) [inline static/js/app.js:306]: Why the double `show()`?

Comment (tim-hellhake) [reply static/js/app.js:306]: I removed the second `show` call.

File: static/js/app.js (hunk 2 above)
Comment (tim-hellhake) [reply static/js/app.js:311]: I added the `catch` accordingly.

File: static/js/extension.js (hunk 1 above)
Comment (tim-hellhake) [reply static/js/extension.js]: Done

**Comment by mrstegeman** [comment]
For anyone seeing this, here's an example for compatibility before and after this change: https://github.com/mozilla-iot/example-extension/commit/97f57a26de4a67738ed3f83919e56a90dfebdfe8

**Pull request merged**